Engine/DataFeeds/UniverseSelection.cs

```diff
@@ -105,6 +106,31 @@ public SecurityChanges ApplyUniverseSelection(Universe universe, DateTime dateTi
                     var fineCollection = new BaseDataCollection();
                     var dataProvider = new DefaultDataProvider();
 
+                    // Create a dictionary of CoarseFundamental keyed by Symbol that also has FineFundamental
+                    // Coarse raw data has SID collision on: CRHCY R735QTJ8XC9X
+                    var coarseData = universeData.Data.OfType<CoarseFundamental>()
+                        .Where(c => c.HasFundamentalData)
+                        .DistinctBy(c => c.Symbol)
+                        .ToDictionary(c => c.Symbol);
+
+                    // Remove selected symbols that does not have fine fundamental data
+                    var anyDoesNotHaveFundamentalData = false;
+                    selectSymbolsResult = selectSymbolsResult
+                        .Where(
+                            symbol =>
+                            {
+                                var result = coarseData.ContainsKey(symbol);
+                                anyDoesNotHaveFundamentalData |= !result;
+                                return result;
+                            }
+                        );
+
+                    if (!_anyDoesNotHaveFundamentalDataWarningLogged && anyDoesNotHaveFundamentalData)
+                    {
+                        Log.Error("Warning: Please do not include securities without fundamental data when performing fine fundamental selection. Check out the documentation for details.");
```

Comment:
Shouldn't this say `...when performing Coarse Fundamental selection` (vs fine)?

Reply:
_algorithm.Debug("Note: Your coarse selection filter was updated to exclude symbols without fine fundamental data. Make sure your coarse filter excludes symbols where HasFundamental is false.");

Please raise to user-level via debug and separate "warning" from "fix" message concepts per example.